test/metabase/cmd/rotate_encryption_key_test.clj

```diff
@@ -19,26 +19,33 @@
    [metabase.util.encryption :as encryption]
    [metabase.util.encryption-test :as encryption-test]
    [metabase.util.i18n :as i18n]
+   [methodical.core :as methodical]
    [toucan.db :as db]
    [toucan.models :as models])
   (:import
    (java.nio.charset StandardCharsets)))
 
 (use-fixtures :once (fixtures/initialize :db))
 
-(defn do-with-model-type
-  [mtype in-type-fns f]
-  (let [type-fns        (var-get #'models/type-fns)
-        before-type-fns @type-fns]
-    (swap! type-fns update mtype merge in-type-fns)
-    (try
-      (f)
-      (finally
-        (reset! type-fns before-type-fns)))))
+(defn- do-with-encrypted-json-caching-disabled
+  [thunk]
+  (let [mf (methodical/add-primary-method
+            @#'models/type-fn
+            [:encrypted-json :out]
+            (fn [_next-method _type _direction]
+              #'mi/encrypted-json-out))]
```

Comment:
Maybe this is another flex but you build multimethods non-destructively and programatically https://github.com/camsaul/methodical#constructing-and-composing-multimethods-programmatically